vllm/model_executor/models/deepseek_v2.py

```diff
@@ -159,11 +159,21 @@ def forward(self, hidden_states: torch.Tensor) -> torch.Tensor:
             shared_output = self.shared_experts(hidden_states)
         # router_logits: (num_tokens, n_experts)
         router_logits, _ = self.gate(hidden_states)
-        final_hidden_states = self.experts(
-            hidden_states=hidden_states,
-            router_logits=router_logits) * self.routed_scaling_factor
+        if hidden_states.dtype != torch.float16:
+            # This is a special case to avoid FP16 overflow
+            final_hidden_states = self.experts(
+                hidden_states=hidden_states,
+                router_logits=router_logits) * self.routed_scaling_factor
+        else:
+            final_hidden_states = self.experts(hidden_states=hidden_states,
+                                               router_logits=router_logits)
         if shared_output is not None:
-            final_hidden_states = final_hidden_states + shared_output
+            if hidden_states.dtype != torch.float16:
+                # This is a special case to avoid FP16 overflow
+                final_hidden_states = final_hidden_states + shared_output
+            else:
+                final_hidden_states = final_hidden_states + shared_output \
+                    * (1. / self.routed_scaling_factor)
```

Comment:
looks like the comment should be a few lines down
```suggestion
            if hidden_states.dtype != torch.float16:
                final_hidden_states = final_hidden_states + shared_output
            else:
                # This is a special case to avoid FP16 overflow
                final_hidden_states = final_hidden_states + shared_output \
                    * (1. / self.routed_scaling_factor)
```

Reply:
Yes.
I have corrected the comment to the right place